src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb

```diff
@@ -535,34 +535,32 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
             Return MakeEndOfLineTrivia(GetNextChar)
         End Function
 
+        Private Function TryGet(num As Integer, ByRef ch As Char) As Boolean
+            Dim ok = CanGet(num)
+            ch = If(ok, Peek(num), ch)
```

Comment:
>ch = If(ok, Peek(num), ch) [](start = 12, length = 26)

Why do an assignment when ```ok``` is false?

Reply:
To make sure it is not left over junk. Setting it to ChrW(0) is safe.

Reply:
> To make sure it is not left over junk. Setting it to ChrW(0) is safe.

Where do we set it to ```ChrW(0```?

---
In reply to: [217792203](https://github.com/dotnet/roslyn/pull/28327#discussion_r217792203) [](ancestors = 217792203)

Reply:
I think it better to only change the contents of `ch` when successful, as  it could be used later in the code.

Reply:
The very first line of code in the parent function sets it to ChrW(0) and if CanGet is false it is just set to itself.

Reply:
Replaced entire routing